x/ibc/core/04-channel/keeper/handshake.go

```diff
@@ -214,6 +225,15 @@ func (k Keeper) ChanOpenAck(
 		return sdkerrors.Wrapf(types.ErrChannelCapabilityNotFound, "caller does not own capability for channel, port ID (%s) channel ID (%s)", portID, channelID)
 	}
 
+	// empty-string is a sentinel value for "allow any identifier" to be selected by
+	// the counterparty channel
+	if counterpartyChannelID != channel.Counterparty.ChannelId && channel.Counterparty.ChannelId != "" {
```

Comment:
ditto
```suggestion
	if channel.Counterparty.ChannelId != "" && counterpartyChannelID != channel.Counterparty.ChannelId {
```

Reply:
Also applied to the spec in https://github.com/cosmos/ics/commit/890e3d0c534ed46b97c794fa0947022380815dbe